ENH: QA/QC of fmriprep outputs (#386)
TheAxonLab/hcph-sops: docs/processing/preprocessing.md

{% include 'code/fmriprep/ss-fmriprep.sh' %}
{% endfilter %}
        ```

??? warning "Not all *fMRIPrep* derivatives were generated"

    - [ ] If some derivatives are missing, it is a sign that *fMRIPrep* encountered an error. Identify what failed in the "Errors" section of the visual report or in the log of the *fMRIPrep* run.
    - [ ] If you could find the solution to the problem, re-run *fMRIPrep* on that particular subject implementing this solution.
    - [ ] In case you do not understand the error message, search for associated keywords in the [issues of the *fMRIPrep* github repository](https://github.com/nipreps/fmriprep/issues); it is likely someone else experienced the same problem before you and reported it. The solution might be documented in the issue. Don't forget to check closed issues!
    - [ ] If that did not help, you might find help on [NeuroStars](https://neurostars.org/).

>     - [ ] If no issue has been opened regarding this error and you did not find answer in NeuroStars, then open an issue in the *fMRIPrep* github repository. Your description of the problem need to be as complete and detailed as possible to help the maintainers identify the problem efficiently.

Reviewer: Some convention for new lines
```suggestion
    - [ ] In case you do not understand the error message, search for associated keywords in the [issues of the *fMRIPrep* github repository](https://github.com/nipreps/fmriprep/issues); it is likely someone else experienced the same problem before you and reported it.
    The solution might be documented in the issue.
    Don't forget to check closed issues!
    - [ ] If that did not help, you might find help on [NeuroStars](https://neurostars.org/).
    - [ ] If no issue has been opened regarding this error and you did not find answer in NeuroStars, then open an issue in the *fMRIPrep* github repository.
    Your description of the problem need to be as complete and detailed as possible to help the maintainers identify the problem efficiently.
```

    - [ ] In case *fMRIPrep* repeatedly fails on that particular session even with the help of the maintainers and even after trying several solutions, exclude that session.

## Visualizing *fMRIPrep*'s individual reports

> Following our protocols<sup>[1]</sup>, the quality of unprocessed images MUST be assessed before after preprocessing to verify it did not go awry.

Reviewer: Or maybe you intended to write "after"
```suggestion
Following our protocols<sup>[1]</sup>, the quality of unprocessed images MUST be assessed before and after preprocessing to verify it did not go awry.
```

- [ ] Scroll down to the next mosaic (surface reconstruction) and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#surface-reconstruction).
- [ ] Scroll down to the functional section, starting with the textual summary and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#textual-summary).
- [ ] Next, apply the [exclusion criteria](qaqc-criteria-preprocessed.md#t2-map) for the T2* map mosaic.
- [ ] Scroll down to the histogram of T2* gray-matter values and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#t2-gray-matter-values).
- [ ] Visualize the dynamic visualization of the co-registration by hovering your mouse on the image to make it toggle and removing it from the image for pausing. Apply the [exclusion criteria](qaqc-criteria-preprocessed.md#alignment-of-functional-and-anatomical-mri-data).
- [ ] Visualize the next mosaic and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#brain-mask-and-anatomicaltemporal-compcor-rois).
- [ ] Skip to the BOLD summary and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#bold-summary).

> - [ ] Scroll down to the heatmap and use it to [choose the regressors](qaqc-criteria-preprocessed.md#correlations-among-nuisance-regressors) you will include in the nuisance regression model.

Reviewer: I added `\t` to separate anatomical and functional QC - feel free to discard if not relevant
```suggestion
- [ ] Open each *fMRIPrep* report on a current Web Browser (*Google Chrome* is preferred).
- [ ] Visualize the first section entitled `Summary` and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#summary).
  - [ ] Scroll down to the next section entitled `Anatomical conformation` and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#anatomical-conformation).
  - [ ] Scroll down to the first mosaic (brain mask and brain tissue segmentation) and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#brain-mask-and-brain-tissue-segmentation-of-the-t1w).
  - [ ] Scroll down to the dynamic visualization of the spatial normalization and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#spatial-normalization-of-the-anatomical-t1w-reference).
  - [ ] Scroll down to the next mosaic (surface reconstruction) and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#surface-reconstruction).
- [ ] Scroll down to the functional section, starting with the textual summary and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#textual-summary).
  - [ ] Next, apply the [exclusion criteria](qaqc-criteria-preprocessed.md#t2-map) for the T2* map mosaic.
  - [ ] Scroll down to the histogram of T2* gray-matter values and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#t2-gray-matter-values).
  - [ ] Visualize the dynamic visualization of the co-registration by hovering your mouse on the image to make it toggle and removing it from the image for pausing. Apply the [exclusion criteria](qaqc-criteria-preprocessed.md#alignment-of-functional-and-anatomical-mri-data).
  - [ ] Visualize the next mosaic and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#brain-mask-and-anatomicaltemporal-compcor-rois).
  - [ ] Skip to the BOLD summary and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#bold-summary).
- [ ] Scroll down to the heatmap and use it to [choose the regressors](qaqc-criteria-preprocessed.md#correlations-among-nuisance-regressors) you will include in the nuisance regression model.
```

{% include 'code/fmriprep/ss-fmriprep.sh' %}
{% endfilter %}
        ```

> ??? warning "Not all *fMRIPrep* derivatives were generated"

Reviewer 2: ```suggestion
??? warning "Not all *fMRIPrep* derivatives were generated"
```

>     - [ ] If some derivatives are missing, it is a sign that *fMRIPrep* encountered an error. Identify what failed in the "Errors" section of the visual report or in the log of the *fMRIPrep* run.

Reviewer 2: ```suggestion
    If some derivatives are missing, it is a sign that *fMRIPrep* encountered an error.
    
    - [ ] Check the "Errors" section of the visual report.
    - [ ] Check the `log/` folder corresponding to the *fMRIPrep* run, carefully ensuring no errors were missed out on the reports.
```

    - [ ] If you could find the solution to the problem, re-run *fMRIPrep* on that particular subject implementing this solution.
    - [ ] In case you do not understand the error message, search for associated keywords in the [issues of the *fMRIPrep* github repository](https://github.com/nipreps/fmriprep/issues); it is likely someone else experienced the same problem before you and reported it.
    The solution might be documented in the issue.

>     Don't forget to check closed issues!

Reviewer 2: ```suggestion
    - [ ] Search for associated keywords in the [issues on *fMRIPrep*'s GitHub repository](https://github.com/nipreps/fmriprep/issues) if the solution remains unclear after the first assessment; it is likely someone else experienced the same problem before you and reported it.
        The solution might be documented in the issue.
        Don't forget to check closed issues!
```

>     - [ ] If that did not help, you might find help on [NeuroStars](https://neurostars.org/).

Reviewer 2: ```suggestion
    - [ ] Search for the issue on [NeuroStars](https://neurostars.org/), if the solution remains unclear.
```

>     - [ ] If no issue has been opened regarding this error and you did not find answer in NeuroStars, then open an issue in the *fMRIPrep* github repository.
>     Your description of the problem need to be as complete and detailed as possible to help the maintainers identify the problem efficiently.

Reviewer 2: ```suggestion
    - [ ] Open an issue in *fMRIPrep*'s GitHub repository to report the problem, if the solution remains elusive.
        Your description of the problem needs to be as complete and detailed as possible to help the maintainers identify the problem efficiently.
```

>     - [ ] In case *fMRIPrep* repeatedly fails on that particular session even with the help of the maintainers and even after trying several solutions, exclude that session.

Reviewer 2: ```suggestion
    - [ ] Re-run *fMRIPrep* on that particular subject and session after implementing a solution.
    
    !!! danger "If the error remains despite all efforts, the session MAY be excluded"

```

> 

Reviewer 2: ```suggestion
```

## Visualizing *fMRIPrep*'s individual reports

Following our protocols<sup>[1]</sup>, the quality of unprocessed images MUST be assessed before and after preprocessing to verify it did not go awry.

- [ ] Open each *fMRIPrep* report on a current Web Browser (*Google Chrome* is preferred).
- [ ] Visualize the first section entitled `Summary` and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#summary).

    !!! warning "Running through the visual report does not have to be executed in a fixed order"

        While we describe the run through the report as linear, often inspecting other reportlets can help make a decision about the exclusion criteria related to the reportlet at hand. 
        As such, we encourage you to jump back and forth between visualizations as much as needed.

> - [ ] Scroll down to the next section entitled `Anatomical conformation` and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#anatomical-conformation).

Reviewer 2: ```suggestion

### Anatomical preprocessing assessment

- [ ] Assess the "Anatomical conformation" section and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#anatomical-conformation).
```

> - [ ] Scroll down to the first mosaic (brain mask and brain tissue segmentation) and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#brain-mask-and-brain-tissue-segmentation-of-the-t1w).

Reviewer 2: ```suggestion
- [ ] Assess the mosaic showing the calculated brain mask and brain tissue segmentation, and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#brain-mask-and-brain-tissue-segmentation-of-the-t1w).
```

> - [ ] Scroll down to the dynamic visualization of the spatial normalization and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#spatial-normalization-of-the-anatomical-t1w-reference).

Reviewer 2: ```suggestion
- [ ] Visualize the spatial normalization flickering mosaic, and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#spatial-normalization-of-the-anatomical-t1w-reference).
```

> - [ ] Scroll down to the next mosaic (surface reconstruction) and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#surface-reconstruction).

Reviewer 2: ```suggestion
- [ ] Assess the surface reconstruction mosaic, and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#surface-reconstruction).
```

> - [ ] Scroll down to the functional section, starting with the textual summary and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#textual-summary).

Reviewer 2: ```suggestion
### Assessment of fMRI Preprocessing

- [ ] Scrutinize the textual summary and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#textual-summary).
```

> - [ ] Next, apply the [exclusion criteria](qaqc-criteria-preprocessed.md#t2-map) for the T2* map mosaic.

Reviewer 2: ```suggestion
- [ ] Visualize the T2* map mosaic, and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#t2-map).
```

> - [ ] Scroll down to the histogram of T2* gray-matter values and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#t2-gray-matter-values).

Reviewer 2: ```suggestion
- [ ] Check the T2* gray-matter intensity histogram, and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#t2-gray-matter-values).
```

> - [ ] Visualize the dynamic visualization of the co-registration by hovering your mouse on the image to make it toggle and removing it from the image for pausing. Apply the [exclusion criteria](qaqc-criteria-preprocessed.md#alignment-of-functional-and-anatomical-mri-data).

Reviewer 2: ```suggestion
- [ ] Visualize the dynamic visualization of the co-registration, apply the [exclusion criteria](qaqc-criteria-preprocessed.md#alignment-of-functional-and-anatomical-mri-data).
    Flickering between T1w and BOLD images is active while hovering your mouse on the mosaic area.
```

> - [ ] Visualize the next mosaic and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#brain-mask-and-anatomicaltemporal-compcor-rois).

Reviewer 2: ```suggestion
- [ ] Visualize the next mosaic displaying regions of interest (ROIs) where nuisance series are estimated, and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#brain-mask-and-anatomicaltemporal-compcor-rois).
```

> - [ ] Skip to the BOLD summary and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#bold-summary).

Reviewer 2: ```suggestion
- [ ] Visualize the carpet plot and nuisance signals panel, and apply the [exclusion criteria](qaqc-criteria-preprocessed.md#bold-summary).
```

> - [ ] Scroll down to the heatmap and use it to [choose the regressors](qaqc-criteria-preprocessed.md#correlations-among-nuisance-regressors) you will include in the nuisance regression model.

Reviewer 2: ```suggestion
- [ ] Visualize the confound correlation heatmap and use it to [choose the regressors](qaqc-criteria-preprocessed.md#correlations-among-nuisance-regressors) you will include in the nuisance regression model.
```

> - [ ] Verify that no error is reported in the "Errors" section. 

Reviewer 2: ```suggestion
- [ ] Verify that no errors are reported within the "Errors" section. 

```

> If there is, follow the procedure described in the appendage "Not all *fMRIPrep* derivatives were generated".

Reviewer 2: ```suggestion
If errors or quality issues are encountered, find the issue corresponding to that session in the dataset's repository and report a comprehensive description of the problems.
In case of *fMRIPrep* failure, follow the procedure described above "Not all *fMRIPrep* derivatives were generated".
```
